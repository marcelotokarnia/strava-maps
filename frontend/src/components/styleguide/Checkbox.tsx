import React, { Component } from 'react'

interface CheckboxProps {
  className: string
  text: string
  onChange?: (value: boolean) => any
  disabledText?: string
  checked: boolean
}

export default class Checkbox extends Component<CheckboxProps> {
  state = {
    checked: this.props.checked,
  }
  handleInputChange = () => {
    this.setState(
      {
        checked: !this.state.checked,
      },
      () => {
        this.props.onChange && this.props.onChange(this.state.checked)
      }
    )
  }

  render() {
    return (
      <div className={this.props.className}>
        <div className="dib mr2 helvetica silver">
          {this.state.checked ? this.props.text : this.props.disabledText || this.props.text}
        </div>
        <div className="relative dib">
          <input
            className="absolute z-5 w-100 h-100 o-0 pointer"
            type="checkbox"
            checked={this.props.checked}
            onChange={this.handleInputChange}
          />
          <div
            className={`relative z-4 dib w3 h2 bg-mid-gray overflow-hidden br4 v-mid bg-animate ${
              this.state.checked ? 'bg-green' : ''
            }`}
          >
            <div
              className={`absolute right-auto left-0 w2 h2 br4 bg-silver shadow-4 t-cb bg-animate ${
                this.state.checked ? 'left-2 bg-moon-gray' : ''
              }`}
            />
          </div>
        </div>
      </div>
    )
  }
}
