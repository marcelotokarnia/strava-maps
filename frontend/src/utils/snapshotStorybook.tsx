import { BrowserRouter } from 'react-router-dom'
import { dissoc } from 'ramda'
import React from 'react'
import renderer from 'react-test-renderer'

export default (stories: any) => {
  return describe(stories.default.title, () => {
    Object.keys(dissoc('default', stories)).forEach(name =>
      it(name, () => {
        const tree = renderer.create(<BrowserRouter>{stories[name]()}</BrowserRouter>).toJSON()
        expect(tree).toMatchSnapshot()
      })
    )
  })
}
