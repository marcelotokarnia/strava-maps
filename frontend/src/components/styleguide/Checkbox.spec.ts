// THIS FILE IS AUTOGENERATED; DO NOT EDIT MANUALLY
import * as tests from './Checkbox.stories'
import snapshotStorybook from '../../utils/snapshotStorybook'

snapshotStorybook(tests)
