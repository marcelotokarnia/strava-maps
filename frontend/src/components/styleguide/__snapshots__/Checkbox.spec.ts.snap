// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Styleguide Checkbox Component checked 1`] = `
<div>
  <div
    className="dib mr2 helvetica silver"
  >
    Click to Disable
  </div>
  <div
    className="relative dib"
  >
    <input
      checked={true}
      className="absolute z-5 w-100 h-100 o-0 pointer"
      onChange={[Function]}
      type="checkbox"
    />
    <div
      className="relative z-4 dib w3 h2 bg-mid-gray overflow-hidden br4 v-mid bg-animate bg-green"
    >
      <div
        className="absolute right-auto left-0 w2 h2 br4 bg-silver shadow-4 t-cb bg-animate left-2 bg-moon-gray"
      />
    </div>
  </div>
</div>
`;

exports[`Styleguide Checkbox Component notChecked 1`] = `
<div>
  <div
    className="dib mr2 helvetica silver"
  >
    Click to Enable
  </div>
  <div
    className="relative dib"
  >
    <input
      className="absolute z-5 w-100 h-100 o-0 pointer"
      onChange={[Function]}
      type="checkbox"
    />
    <div
      className="relative z-4 dib w3 h2 bg-mid-gray overflow-hidden br4 v-mid bg-animate "
    >
      <div
        className="absolute right-auto left-0 w2 h2 br4 bg-silver shadow-4 t-cb bg-animate "
      />
    </div>
  </div>
</div>
`;
